### DISCUSSION TITLE
Solar PV divert

### DISCUSSION THREAD

**Discussion by rcmad** [firstof9/openevse#375] [Q&A]
Hi. I have just got this integration up and running and want to setup the solar divert part. I have a sensor in homassistant that pulls house load in watts
![image](https://github.com/user-attachments/assets/b8ba5916-6820-405f-b180-2ee8726c6675)

I have added that into the openevse setup

![image](https://github.com/user-attachments/assets/e1b599f9-d82f-4f9d-83bc-68ae485542cd)

But I am showing unavailable for PV available current.

![image](https://github.com/user-attachments/assets/2469fc25-c3cb-4748-b7c7-72f640d63ae5)

What have I got wrong with the config?

**Comment by firstof9** [comment]
You have no solar production sensor setup.

**Reply by rcmad** [reply]
Doh!

**Reply by rcmad** [reply]
I did actually have the same sensor in the solar sensor setup, so I still have something else wrong.

**Reply by firstof9** [reply]
Once you have a solar production sensor or grid import/export sensor set, if you toggle on PV divert in eco mode it'll charge by only what solar output is allowing.

**Comment by thekeymeister** [comment]
I've just been trying to get this working myself, I got there in the end, the bit that was missing was changing mode to "Export" in the Openevse's own config page under Self Production. Before that, no matter what I tried, the OpenEVSE PV Available Current was always showing 0A. Probably my own misunderstanding of the device, but thought I should post in case it helps someone else get it working. 

![Screenshot_20250612-170349](https://github.com/user-attachments/assets/e7716680-e530-4967-bdf7-9e21c16e2555)



Not sure if it's a bug with this integration or my misunderstanding again, I noticed that once you reconfigure to add the sensors, it's impossible to remove them, you can only replace them with others. If you remove them it seems to accept it when you press submit, however when you go to reconfigure the old values are still there.  This made testing a little bit difficult!

**Comment by zanderz** [comment]
Thanks for the tips, I got this working too after I realized I had to set the sensor units to W instead of kW

**Comment by zanderz** [comment]
I had some more problems with this recently and I think I have partially figured it out. Sometimes in the evening after sunset the car would charge at max power until the house battery was depleted to the reserve. I saw that this integration recently switched to web apis instead of mqtt so I thought it was being confused by the mix of both, but that wasn't it. It turns out that if you set the tesla powerwall3 to the "savings" operation mode instead if "self powered", it will sometimes send power to the grid from the battery during peak times. Of course this makes OpenEVSE think there is spare power so it sends it to the car.

I am thinking of trying a helper to combine the minimum of solar production and grid export instead of just grid export, so the powerwall can export in the evening without the car soaking it up. Is there another approach I haven't considered?

Also BTW I totally turned off all MQTT signals in and out of OpenEVSE and this integration still works, so all the advice about mqtt settings seems outdated if I understand that correctly.

In case it wasn't clear, my intent is to make sure the house backup is fully charged before charging the car from solar.
